test/test_prototype_builtin_datasets.py

```diff
@@ -0,0 +1,99 @@
+import functools
+import io
+
+import builtin_dataset_mocks
+import pytest
+from torchdata.datapipes.iter import IterDataPipe
+from torchvision.prototype import datasets
+from torchvision.prototype.datasets.utils._internal import sequence_to_str
+
+
+_loaders = []
+_datasets = []
+
+# TODO: this can be replaced by torchvision.prototype.datasets.list() as soon as all builtin datasets are supported
+TMP = [
+    "mnist",
+    "fashionmnist",
+    "kmnist",
+    "emnist",
+    "qmnist",
+    "cifar10",
+    "cifar100",
+    "caltech256",
+    "caltech101",
+    "imagenet",
+]
+for name in TMP:
+    loader = functools.partial(builtin_dataset_mocks.load, name)
+    _loaders.append(pytest.param(loader, id=name))
+
+    info = datasets.info(name)
+    _datasets.extend(
+        [
+            pytest.param(*loader(**config), id=f"{name}-{'-'.join([str(value) for value in config.values()])}")
+            for config in info._configs
+        ]
+    )
+
+loaders = pytest.mark.parametrize("loader", _loaders)
+builtin_datasets = pytest.mark.parametrize(("dataset", "mock_info"), _datasets)
+
+
+class TestCommon:
+    @builtin_datasets
+    def test_smoke(self, dataset, mock_info):
+        if not isinstance(dataset, IterDataPipe):
+            raise AssertionError(f"Loading the dataset should return an IterDataPipe, but got {type(dataset)} instead.")
+
+    @builtin_datasets
+    def test_sample(self, dataset, mock_info):
```

Comment:
I feel like it is reasonable to assume that if the first sample is valid, all the other will also be. Otherwise we would need to have some weird branching within the dataset that could cause this and this should be caught during code review.

Of course we can also  merge `test_sample` and `test_num_samples` to check the validity of every sample. But then the question arises, if we need to check all samples for every single test? For example, what about the decoding tests? Should we also check every sample there? If not, what is the difference to `test_sample`?

Reply:
I feel like it's hard to do these without clogging up the CI. I'd say the first one is ok for 90% of the cases, but I think we should re-think these for the video datasets with clips to include somehow randomly drawn sample bc these can be a bit unstable.

Reply:
If you are thinking about the video utilities you added in #4838, I don't think we need to change anything here. Since they are separate from the datasets, we can also test them separately.